Libraries/LibWeb/DOM/EventTarget.cpp

```diff
@@ -822,10 +824,19 @@ bool EventTarget::dispatch_event(Event& event)
         if (event.type() == UIEvents::EventNames::mousedown)
             return true;
 
-        // FIXME:
         // pointerdown, provided the event's pointerType is "mouse".
+        if (event.type() == UIEvents::EventNames::pointerdown) {
+            if (auto* pointer_event = as_if<UIEvents::PointerEvent>(&event))
+                return pointer_event->pointer_type() == "mouse";
+        }
+
         // pointerup, provided the event's pointerType is not "mouse".
-        // touchend.
+        if (event.type() == UIEvents::EventNames::pointerup) {
+            if (auto* pointer_event = as_if<UIEvents::PointerEvent>(&event))
+                return pointer_event->pointer_type() != "mouse";
```

Comment:
Do we really store the pointer type as a string? I suppose we do. Though the way https://w3c.github.io/pointerevents/#dom-pointerevent-pointertype is written it seems like we could have it as an optional enum :shrug: . Presumably there's a reason @shannonbooth implemented it that way in https://github.com/SerenityOS/serenity/pull/23886

for this PR, the comment is "prefer to compare to StringView literals". `return pointer_event->pointer_type != "mouse"sv;`

Reply:
Its a string in IDL is why

Reply:
we could probably have an enum version of it, but a string conversion needs to exist for javascript integration in any case

Reply:
For the web platform test that triggered this faulty behavior, for the test to succeed it's actually the pointerUp + != "mouse" that makes sure that the user activation is set, so to speak (also, it's of `Document` type), so I don't know sure if the `pointer_type` ever has a value other than `""` in ladybird.

Either way, this probably should be decided in a follow up, or somewhere else though, right?

Reply:
> this should be a follow-up

Indeed, that's why my first commented ended with "but for this PR, only worry about this nitpick"

Reply:
> > this should be a follow-up
> 
> Indeed, that's why my first commented ended with "but for this PR, only worry about this nitpick"

Cool! Ok, I'll mark this as resolved

Reply:
Would be a good to actually apply the string view literal change I requested though 😅

Reply:
damn lol :laughing:, of course!